test/server/admin/stats_handler_test.cc

```diff
@@ -1069,6 +1071,115 @@ TEST_P(AdminStatsTest, SortedHistograms) {
   }
 }
 
+// Sets up a test using real threads to reproduce a race between deleting scopes
+// and iterating over them.
+class ThreadedTest : public testing::Test {
```

Comment:
Instead of a stress style test (or in addition) is it possible to do a thread synchronizer style test that is deterministic that also fails?

Reply:
I'm looking at it. I definitely don't want it to be lieu of the real-threads stress test. One thing I often find with synchronizers is that once you fix the race, you can't trigger the synchronizer without deadlock, but I'm iterating a bit.

Reply:
Done.